### ISSUE TITLE
Pony: BitField support

### ISSUE THREAD

**Issue by alex** [zacharyvoase/django-postgres#1]
PostgreSQL has awesome support for bit fields, this should support 'em!

**Comment by zacharyvoase** [comment]
Happening.

Also happening:

``` pycon
>>> Model.objects.filter(bitfield__xor='110101')
>>> Model.objects.update(bitfield=B('bitfield') ^ '110101')
```

Et cetera. I like `ExpressionNode` for this kind of stuff.

**Comment by alex** [comment]
Awesome, didn't see it noted in the "future" section, so I figured I'd bring it up.

**Comment by zacharyvoase** [comment]
Yup. Seems easy enough to implement; I may be able to get it done this arvo.

**Comment by zacharyvoase** [comment]
OK. Not that easy to implement.

There's a basic implementation of the `BIT` and `VARBIT` types [here](https://github.com/zacharyvoase/django-postgres/blob/master/django_postgres/bitstrings.py). There’s also a `B()`-expression syntax thing for manipulating bit strings atomically; see [these test cases](https://github.com/zacharyvoase/django-postgres/blob/master/test_project/bitstringtest/tests.py) for a short example.

The biggest problem, currently, is the implementation of custom query predicates based on the bitwise operators. Stuff like `__xor`, which I think would be really useful. Also, I’d love a query aggregate like `popcount()`, but apparently Postgres doesn’t even have that built-in.
